validations.md

```diff
@@ -0,0 +1,102 @@
+# Detailed Validation Breakdown
+
+## Creating a new Entity
+
+### Product 
+
+```python 
+product = Product.new(name="ExampleProduct", code="EXPRD", description="Example Product")
+client.create(product)
+``` 
+
+* Product `code` and `name` should not be empty
```

Comment:
I leave this comment here but it's rather a comment for the whole document.

I really like the fact that we are documenting the validations that we're doing. I'm not sure though if we should do it in a dedicated markdown file like this, or rather as inline docs, closer to the code, and then we generate the documentation in some way (sphinx or similar).